controllers/utils/daemonset.go

```diff
@@ -0,0 +1,71 @@
+package utils
```

Comment:
package, classes, functions, etc. named "utils" should be avoided. I think, as @0sewa0 suggested, "kubeobjects" would be a better fit. A call would then look like "kubeobjects.CreateOrUpdateDaemonSet(...)".